dvc/project.py

```diff
@@ -1128,35 +1160,39 @@ def metrics_show(self,
         raise DvcException(msg)
 
     def _metrics_modify(self, path, typ=None, xpath=None, delete=False):
-        out = self._find_output_by_path(path)
-        if not out:
+        outs = self._find_output_by_path(path)
+
+        if not outs:
             msg = "unable to find file '{}' in the pipeline".format(path)
             raise DvcException(msg)
 
-        if out.scheme != 'local':
-            msg = "output '{}' scheme '{}' is not supported for metrics"
-            raise DvcException(msg.format(out.path, out.path_info['scheme']))
+        for out in outs:
```

Comment:
do we expect modify to deal with multiple metric files as well? I mean, do we actually need a cycle here?

Reply:
Yes, because outs comes from `self._find_output_by_path(path)`, which in case the -R is active and multiple metric files have been found will return a list

Reply:
@tdeboissiere totally! My question was more from a product perspective - what should we do when someone runs `dvc metric modify [something] -R path`. Should we allow that at all or just raise an exception? Should we care for this operation to be atomic? Probably, @efiop has more knowledge on this.

Reply:
@shcheklein is right, since -R is not supported for `metrics_modify` let's remove this cycle and put something like
```
assert(len(outs) == 1)
```
for now.